src/Relay.ts

```diff
@@ -0,0 +1,103 @@
+interface Message {
+  to?: string
+  message?:string
+  payload?:any
+  sender?:string
+}
+
+class Relay extends EventTarget {
+  id;
+  name;
+  connections = new Map();
+
+  constructor(name:string) {
+    super();
+    this.name = name;
+    this.broadcast = this.broadcast.bind(this);
+    this.send = this.send.bind(this);
+    this.addConnection = this.addConnection.bind(this);
+    this.removeConnection = this.removeConnection.bind(this);
+    this.createEvent = this.createEvent.bind(this);
```

Comment:
Instead of binding methods in the constructor, you can use arrow function syntax when defining the method below:
```ts
broadcast: (message: Message, sender: ???) => {
  ...
}
```
That said, if you don't plan on passing the method around as a function, then you don't need to bind it, which means it can live on `Relay.prototype`, rather than getting redefined on each `Relay` object.

Reply:
I can convert these to arrow functions. I wasn't sure what style we preferred but it doesn't matter much to me.